Dapper/SqlMapper.cs

```diff
@@ -2101,11 +2101,11 @@ public static void PackListParameters(IDbCommand command, string namePrefix, obj
                             else
                             {
                                 var sb = GetStringBuilder().Append('(').Append(variableName);
-                                if (!byPosition) sb.Append(1);
+                                if (!byPosition) sb.Append(1); else sb.Append(namePrefix).Append(1).Append(variableName);

```

Comment:
a little confused here - it looks like we'll have `{variableName}{namePrefix}1{variableName}` because of the previous line...? (sorry for delay)

Reply:
Oh yeah, i'll try to explain myself because i'm confused too haha

With the test I added I think that the "problem" is easily to view, in concrete is in this method https://github.com/StackExchange/Dapper/blob/master/Dapper/SqlMapper.cs#L1730 and this line 
```csharp
cmd.CommandText = pseudoPositional.Replace(cmd.CommandText, match =>
```

If you debug the above line you will see that cmd.CommandText contains 

```sql
where s1.value in (?,?,?,?) and s2.value = ?score?
```
And the problem with that is that Replace doesn't iterate around '?' character, so at the end of the method with or without my commit cmd.CommandText is gonna be 

```sql
select s1.value as id, s2.value as score 
                        from string_split('1,2,3,4,5',',') s1, string_split('1,2,3,4,5',',') s2
                        where s1.value in (?,?,?,?) and s2.value = ?
```
but without my commit cmd.Parameters has one parameter, not five because cmd.CommandText with my commit looks like the above image and Replace does its job!

```sql
where s1.value in (?ids1?,?ids2?,?ids3?,?ids4?) and s2.value = ?score?
```

Thanks for all Marc!!